core/primitives/src/shard_layout.rs

```diff
@@ -83,17 +98,201 @@ pub struct ShardLayoutV1 {
     /// Maps shards from the last shard layout to shards that it splits to in this shard layout,
     /// Useful for constructing states for the shards.
     /// None for the genesis shard layout
-    shards_split_map: Option<ShardSplitMap>,
+    shards_split_map: Option<ShardsSplitMap>,
     /// Maps shard in this shard layout to their parent shard
     /// Since shard_ids always range from 0 to num_shards - 1, we use vec instead of a hashmap
     to_parent_shard_map: Option<Vec<ShardId>>,
     /// Version of the shard layout, this is useful for uniquely identify the shard layout
     version: ShardVersion,
 }
 
+impl ShardLayoutV1 {
+    // In this shard layout the accounts are divided into ranges, each range is
+    // mapped to a shard. The shards are contiguous and start from 0.
+    fn account_id_to_shard_id(&self, account_id: &AccountId) -> ShardId {
+        // Note: As we scale up the number of shards we can consider
```

Comment:
nit: note is outdated, in a certain sense
I think no one will add more shards to V1 layout